addons/File_Sender/src/presenter.js

```diff
@@ -0,0 +1,637 @@
+                       headers: {
+                           'Authorization': 'JWT ' + json.token
+                       },
+                       body: formData
+                    })
+               ).then(result => result.json()).then(
+                    success => {
+                        presenter.fireSendFileEvent(success.uploaded_file_id, targetID);
+                        var fileUrl = presenter.fileDownloadEndpointUrl + success.uploaded_file_id;
+                        presenter.setSentFile(file.name, fileUrl);
+                        presenter.showSentFile();
+                    }
+                ).catch((err) => {
+                        console.log(err);
+                    }
+                );
+            });
+        });
+    }
+
+    presenter.executeCommand = function(name, params) {
+        if (!presenter.configuration.isValid) return;
+
+        var commands = {
+            'showTeacherList': presenter.showTeacherList
+        };
+
+        Commands.dispatch(commands, name, params, presenter);
+    };
+
+    presenter.getParagraphModule = function() {
+        if (presenter.configuration.sourceType !== SOURCE_TYPES.PARAGRAPH) return null;
+        var module = presenter.playerController.getModule(presenter.configuration.sourceId);
+        if (module == null || !module.hasOwnProperty('getText')) return null;
+        return module;
+    }
+
+
+    presenter.getTextFile = function() {
+        var module = presenter.getParagraphModule();
+        if (module == null) return null;
+        var text = module.getText();
+        var file = new File([text], "paragraph.html", {type: "text/html;charset=utf-8"});
+        return file;
+    }
+
+    presenter.getMediaRecorderModule = function() {
+        if (presenter.configuration.sourceType !== SOURCE_TYPES.MEDIA_RECORDER) return null;
+        var module = presenter.playerController.getModule(presenter.configuration.sourceId);
+        if (module == null || !module.hasOwnProperty('getMP3File') || !module.hasOwnProperty('isEmpty')) return null;
+        return module;
+    }
+
+    presenter.getMP3File = function() {
+        var module = presenter.getMediaRecorderModule();
+        if (module == null) return null;
+        return module.getMP3File();
+    }
+
+    presenter.getFile = function() {
+        return new Promise(resolve => {
+        if (presenter.configuration.sourceType === SOURCE_TYPES.PARAGRAPH) {
+            resolve(presenter.getTextFile());
+        } else if (presenter.configuration.sourceType === SOURCE_TYPES.FILE) {
+            if (presenter.views.formInput[0].files.length > 0) {
+                 resolve(presenter.views.formInput[0].files[0]);
+            } else {
+                resolve(null);
+            }
+        } else { //SOURCE_TYPES.MEDIA_RECORDER
+            var mp3 = presenter.getMP3File();
+            if (mp3 == null) {
+                resolve(null);
+            } else {
+                mp3.then(file => resolve(file));
```

Comment:
Linijkę 456 można zmienić po prostu na resolve(mp3);
Mały js fiddle, że działa to w ten sam sposób - https://jsfiddle.net/od25buws/
`If the value has a “then” attached to the promise, then the returned promise will follow that “then” to till the final state.`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve
(tyle info bo sam byłem ciekaw jak to działa :P )